pkg/booleanpolicy/policyversion/convertor.go

```diff
@@ -6,40 +6,24 @@ import (
 	"github.com/stackrox/rox/pkg/utils"
 )
 
-// EnsureConvertedToLatest converts the given policy into a Boolean policy, if it is not one already.
+// EnsureConvertedToLatest converts the given policy to the latest version (as defined by CurrentVersion), if it isn't already
+// The policy is modified in place.
 func EnsureConvertedToLatest(p *storage.Policy) error {
 	if p == nil {
 		return errors.New("nil policy")
 	}
-	policyVersion, err := FromString(p.GetPolicyVersion())
-	if err != nil {
-		return err
-	}
 
-	if Compare(policyVersion, Version1()) >= 0 && len(p.GetPolicySections()) == 0 {
-		return errors.New("empty sections")
+	ver, err := FromString(p.GetPolicyVersion())
+	if err != nil || ver.String() == legacyVersion { // Legacy versions (i.e. no version) is no longer supported
```

Comment:
I don't think this `ver.String() == legacyVersion` comparison is required here. This can just be allowed to happen through the error returned by `UpgradePolicyTo` below.